sdk/azcore/arm/resource_identifier.go

```diff
@@ -0,0 +1,315 @@
+//go:build go1.16
+// +build go1.16
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package arm
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+)
+
+const (
+	providersKey             = "providers"
+	subscriptionsKey         = "subscriptions"
+	resourceGroupsLowerKey   = "resourcegroups"
+	locationsKey             = "locations"
+	builtInResourceNamespace = "Microsoft.Resources"
+)
+
+var (
+	RootResourceIdentifier = &ResourceIdentifier{
```

Comment:
Do these really need to be exported?

Reply:
Yeah - this is designed to be the root level parent of the resource identifiers instead of `nil`.
And this is also a resource identifier that you cannot get from `ParseResourceIdentifier`